synapse/util/ratelimitutils.py

```diff
@@ -52,6 +53,34 @@ def new_limiter() -> "_PerHostRatelimiter":
             str, "_PerHostRatelimiter"
         ] = collections.defaultdict(new_limiter)
 
+        # We track the number of affected hosts per time-period so we can
+        # differentiate one really noisy homeserver from a general
+        # ratelimit tuning problem across the federation.
+        LaterGauge(
+            "synapse_rate_limit_sleep_affected_hosts",
+            "Number of hosts that had requests put to sleep",
+            [],
+            lambda: count(
+                bool,
+                [
+                    ratelimiter.should_sleep()
+                    for ratelimiter in self.ratelimiters.values()
+                ],
+            ),
```

Comment:
```suggestion
            lambda: sum(
                1
                for ratelimiter in self.ratelimiters.values()
                if ratelimiter.should_sleep()
            ),
```

I think is a lot clearer? Alternatives include `len([1 for r in ratelimiters if r.should_sleep()])`, but that instantiates a new list

Reply:
Good shout! Both yours and [@reivilibre's](https://github.com/matrix-org/synapse/pull/13541#discussion_r947684275) seem much better! Updated ✅ and we can also follow-up to remove that existing `count` instance in another PR ⏩